Nginx-Certbot.md

```diff
@@ -0,0 +1,181 @@
+because it is easier to configure for first-time users. The wiki page 
+assumes that the machine used to serve content is a Digital Ocean 
+droplet, but the specific steps won't differ widely between services.
+
+## Step 1: Install `nginx`
+
+Installation is relatively easy.
+- ssh into your droplet.
+- To confirm that the software is not currently installed, run the 
+  following command:<br>
+  ```
+  $ which nginx
+  $
+  ```
+  <br>
+  If you see no executable path before the prompt reappears, then you 
+  will need to install `nginx`.
+- Install `nginx` with the following command:<br>
+  ```
+  $ sudo apt-get install nginx
+  ```
+  <br>
+  When prompted with `Do you want to continue? [Y/n]`, type yes and 
+  press return.
+- To confirm that `nginx` was installed, rerun `which nginx`.
+  You should now see a path to the executable in the console.
+
+## Step 2: Get Your Website Code onto the Server
+
+Assuming that your code is in a `git` repository, clone the repository 
+onto your machine.
+
+In our example, we will be serving 
+[this GitHub repository](https://github.com/UWB-ACM/website-example-2) 
+at [phry.me](http://phry.me).
+
+To clone the repository, run the following commands and replace the 
+GitHub URL with the URL to your own repository.
+
+```
+$ cd /var/www
+$ git clone https://www.github.com/UWB-ACM/website-example-2.git
+$ cd website-example-2
+$ pwd
+/var/www/website-example-2
+```
+
+Note: the output for the `pwd` command will be important for the 
+next step.
+
+## Step 3: Set Up Configuration for `nginx`
+
+Now that you have the source code for your website on your machine, 
+you need to point `nginx` to the directory of the source code and 
+ensure that the content gets served on your domain.
+
+The configuration file your your site will be located under `/etc/nginx/sites-available`. On a fresh installation of `nginx`, a 
+site configuration file called `default` is provided as an example.
+
+Run the following commands and replace `phry.me` with the domain name 
+you chose. _Note: this isn't a strict requirement, but is a common 
+naming convention. It is recommended to use naming conventions._
+
+```
+$ cd /etc/nginx/sites-available
+$ ls
+default
+```
+
+Now, in your preferred text editor, create a file named `phry.me` 
+(or whatever your domain name is). Add the following text to the 
+file:
+
+```
+server {
+    # replace the filepath with the output from the pwd command from step 2
+    root /var/www/website-example-2;
+    # the file name should be the desired homepage file for your website located in the directory named in the previous line
+    index index.html;       
+    # this line is for the domain name you chose
+    # make sure the domain name matches the one you bought!
+    server_name phry.me lol.phry.me;
+
+    # these lines will help protect users from accessing raw files in your repository folder
+    location / {
+        try_files $uri $uri/ =404;
```

Comment:
this would protect against leaking a `.git` dir?

Reply:
turns out i didn't fully understand what these lines were for :sweat_smile: 

`location / {` tells nginx to parse the specified URI from the root directory defined in the `server{}` block. `try_files` tells nginx to look first for the page matching URI exactly, then a directory matching URI, and if those fail, to exactly redirect to `404`. Once again, Digital Ocean docs are great, and a more in-depth explanation is [here](https://www.digitalocean.com/community/tutorials/understanding-nginx-server-and-location-block-selection-algorithms).

Reply:
I am noodling with `chmod` permissions for files under `.git` but haven't found anything that prevents other users from reading files under `.git`. Specifically, I tried `chmod o-rx *` in the `.git` folder to no avail (so you can go to https://phry.me/.git/HEAD and download the file)

Reply:
@Chris-Johnston I'm going to add this to the markdown file, but am reluctant to write an in-depth explanation of what it does beyond a simple comment

Reply:
Addressed in 8fef2fa52085214ae4fb3b0e296c74733c0121f2

Reply:
Sure, I don't think it's necessary to cover this in-depth either.